### PR TITLE
Add spase crosswalk

### PR DESCRIPTION
**Pull request by BaptisteCecconi**
Following the presentation about CodeMeta by @dgarijo at the DASH meeting (heliophysics community) this morning, I prepared a first draft for a crosswalk between CodeMeta and SPASE (Space Physics Extract Search and Extract). 

SPASE is an information model (a schema) to describe resources in heliophysics. SPASE resource types are Person, Instrument, Observatory, NumericalData, Repository, Software, etc. 

The full schema is available here: https://spase-group.org/data/index.html

It is serialised as an XML schema (see [here](https://spase-group.org/data/model/index.html) for the various versions). 

An interactive web view of the Software class of the schema is available [here](https://spase-group.org/data/model/spase-2.6.1/spase-2_6_1_xsd.html#http___www.spase-group.org_data_schema_Spase_Software).

I copy below the list of metadata element for the Software class in a flat view:
```
ResourceID
ResourceHeader
	ResourceName
	AlternateName
	DOI
	ReleaseDate
	RevisionHistory
		RevisionEvent
	ExpirationDate
	Description
	Acknowledgment
	PublicationInfo
		Title
		Authors
			Author
		PublicationDate
		PublishedBy
		LandingPageURL
	Funding
		Agency
		Project
		AwardNumber
	Contact
		PersonID
		Role
		StartDate
		StopDate
		Note
	InformationURL
		Name
		URL
		Description
		Language
	Association
		AssociationID
		AssociationType
		Note
	PriorID
AccessInformation
	RepositoryID
	Availability
	AccessRight
	AccessURL
		Name
		URL
		Style
		ProductKey
		Description
		Language
	Format
	Encoding
	AccessDirectoryTemplate
	AccessFilenameTemplate
	DataExtent
		Quantity
		Units
		Per
	Acknowledgment
SoftwareVersion
ApplicationInterface
CodeLanguage
Prerequisites
ExecutionEnvironment
	OperatingSystems
	Installer
		Availability
		AccessRight
		Acknowledgment
		URL
	Cores
	Storage
	Memory
InputProperty
	Name
	Description
	Caveats
	Units
	ValidMin
	ValidMax
OutputProperty
	Name
	Description
	Caveats
	Units
	ValidMin
	ValidMax
```

And here is an example of SPASE record for a Software:

https://hpde.io/CCMC/Software/Kamodo.html

### CODE REVIEW

File: crosswalks/spase-v2.6.1.csv [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+targetProduct,
+applicationCategory,
+applicationSubCategory,
+downloadUrl,AccessURL/URL
+fileSize,
+installUrl,ExecutionEnvironment/Installer/URL
+memoryRequirements,ExecutionEnvironment/Memory
+operatingSystem,OperatingSystem
+permissions,
+processorRequirements,ExecutionEnvironment/Cores
```
Comment (progval) [inline crosswalks/spase-v2.6.1.csv:15]: Doesn't seem to match, see https://schema.org/processorRequirements while https://spase-group.org/data/model/spase-2.6.1/index.html has type `xsd:double` (presumably that's a number of cores)

Comment (BaptisteCecconi) [reply crosswalks/spase-v2.6.1.csv]: You're right, I removed this mapping

**Review by progval — APPROVED** [review]

**Comment by dgarijo** [comment]
Thanks for your rapid response, @progval. @BaptisteCecconi asked me to review this, and I would like to make sure we are not missing other predicates. Please let me have a look this week and comment here. Thanks!!

**Comment by rebeccaringuette** [comment]
@dgarijo Thanks for taking a look at this. One critical problem in SPASE is that external vocabularies are not supported, which is one of many reasons we cannot use SPASE as the metadata structure for the Heliophysics Software Search Interface. Perhaps the structure we design for that search interface will be what is used in SPASE 3.0 for software, but that remains to be seen. Because of those problems, I don't see much value in adding SPASE to CodeMeta's crosswalks, but it would be useful to see what other concepts are missing for software in SPASE. I expect to be in contact once we have our structure drafted in the coming months.

**Comment by BaptisteCecconi** [comment]
Hi @rebeccaringuette, I agree with your statements. The crosswalk I initiated should really be seen as a way to see what is missing on our side, as you say.

File: crosswalks/spase-v2.6.1.csv [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+identifier,ResourceHeader/DOI
+name,
+address,
+reviewAspect,
+reviewBody,
+endDate,
+roleName,
+startDate,
+softwareSuggestions,
+maintainer,
```
Comment (dgarijo) [inline crosswalks/spase-v2.6.1.csv:64]: this should be ResourceHeader/Contact (http://www.spase-group.org/data/schema/Contact)

File: crosswalks/spase-v2.6.1.csv [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+version,
+isAccessibleForFree,
+isPartOf,
+hasPart,
+position,
+description,ResourceHeader/Description
+identifier,ResourceID
+name,ResourceHeader/ResourceName
+sameAs,
+url,
```
Comment (dgarijo) [inline crosswalks/spase-v2.6.1.csv:49]: Shouldn't this be ResourceHeader/InormationURL/URL ?

File: crosswalks/spase-v2.6.1.csv [hunk 4]
```diff
@@ -0,0 +1,74 @@
+Property,SPASE
+codeRepository,RepositoryID
```
Comment (dgarijo) [inline crosswalks/spase-v2.6.1.csv:2]:
```suggestion
codeRepository, AccessInformation/RepositoryID
```

File: crosswalks/spase-v2.6.1.csv [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+downloadUrl,AccessURL/URL
+fileSize,
+installUrl,ExecutionEnvironment/Installer/URL
+memoryRequirements,ExecutionEnvironment/Memory
+operatingSystem,OperatingSystem
+permissions,
+processorRequirements,
+releaseNotes,
+softwareHelp,
+softwareRequirements,
```
Comment (dgarijo) [inline crosswalks/spase-v2.6.1.csv:18]:
```suggestion
softwareRequirements, Prerequisites
```

File: crosswalks/spase-v2.6.1.csv [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+review,
+runtimePlatform,
+targetProduct,
+applicationCategory,
+applicationSubCategory,
+downloadUrl,AccessURL/URL
+fileSize,
+installUrl,ExecutionEnvironment/Installer/URL
+memoryRequirements,ExecutionEnvironment/Memory
+operatingSystem,OperatingSystem
```
Comment (dgarijo) [inline crosswalks/spase-v2.6.1.csv:13]:
```suggestion
operatingSystem,ExecutionEnvironment/OperatingSystems
```

File: crosswalks/spase-v2.6.1.csv [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+contributor,
+copyrightHolder,
+copyrightYear,
+dateCreated,
+dateModified,
+datePublished,ResourceHeader/ReleaseDate
+editor,
+encoding,AccessInformation/Encoding
+fileFormat,AccessInformation/Format
+funder,ResourceHeader/Funding
```
Comment (dgarijo) [inline crosswalks/spase-v2.6.1.csv:33]:
```suggestion
funder,ResourceHeader/Funding/Agency
```

File: crosswalks/spase-v2.6.1.csv (hunk 3 above)
Comment (dgarijo) [inline crosswalks/spase-v2.6.1.csv:49]:
```suggestion
url,ResourceHeader/InormationURL/URL
```

File: crosswalks/spase-v2.6.1.csv (hunk 2 above)
Comment (dgarijo) [inline crosswalks/spase-v2.6.1.csv:64]:
```suggestion
maintainer,ResourceHeader/Contact 
```

File: crosswalks/spase-v2.6.1.csv [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+endDate,
+roleName,
+startDate,
+softwareSuggestions,
+maintainer,
+continuousIntegration,
+buildInstructions,
+developmentStatus,
+embargoEndDate,
+funding,ResourceHeader/Funding
```
Comment (dgarijo) [inline crosswalks/spase-v2.6.1.csv:69]: ResourceHeader/Funding/AwardNumber

**Review by dgarijo — CHANGES_REQUESTED** [review]
Hi @BaptisteCecconi can you please check my suggestions?
I completed them based on the information and definitions in https://spase-group.org/data/model/spase-2.6.1/index.html

**Comment by dgarijo** [comment]
@BaptisteCecconi some of my suggestions are duplicated with comments in text, sorry. I started doing suggestions halfway through since you can commit them if you like them

**Comment by dgarijo** [comment]
@progval please have a quick look to see whether you approve with my suggestions whenever you have the chance. Sorry it took a bit longer than expected.

File: crosswalks/spase-v2.6.1.csv [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+name,ResourceHeader/ResourceName
+sameAs,
+url,
+relatedLink,
+givenName,
+familyName,
+email,
+affiliation,
+identifier,ResourceHeader/DOI
+name,
```
[the remark this answers is not in the document]
Comment (BaptisteCecconi) [reply crosswalks/spase-v2.6.1.csv:56]: not sure. I checked the other cross walks, and there are duplicate `name` entries in the CSV (see, e.g., DataCite.csv). Line 47 is the resource name, while line 56 is the creator name. That might be an issue, indeed.

**Comment by dgarijo** [comment]
Ok, feel free to ignore this then

**Comment by dgarijo** [comment]
What about the other suggestions?

File: crosswalks/spase-v2.6.1.csv (hunk 9 above)
[the remark this answers is not in the document]
Comment (dgarijo) [reply crosswalks/spase-v2.6.1.csv:56]: Feel free to ignore this one then

**Comment by dgarijo** [comment]
Thanks @BaptisteCecconi, if you have questions/comments on the remaining ones let me know

**Comment by BaptisteCecconi** [comment]
I update all fields according to your comments. The only remaining issue may be the "funding" term, which could be mapped to 2 entities in SPASE (as you propose). I put a human readable process in the CSV (probably not a good thing for you :-)

**Review by dgarijo — APPROVED** [review]
I leave this to @progval to decide. I think the line `funding,ResourceHeader/Funding/AwardNumber or ResourceHeader/Funding/Project (if no AwardNumber)` will likely cause problems. It's a tricky one.

**Comment by progval** [comment]
In other crosswalks (SPDX, Leiningen, NodeJS, Cabal, CFF) we use ` / ` instead of ` or ` but it would be confusing here, so let's keep ` or ` and change the other crosswalks.

No need for the parenthesis at the end, IMO. Translations to Codemeta can have both values.

**Comment by dgarijo** [comment]
@progval if no further changes are required, can you merge this PR?

File: crosswalks/spase-v2.6.1.csv [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+endDate,
+roleName,
+startDate,
+softwareSuggestions,
+maintainer,ResourceHeader/Contact
+continuousIntegration,
+buildInstructions,
+developmentStatus,
+embargoEndDate,
+funding,ResourceHeader/Funding/AwardNumber or ResourceHeader/Funding/Project (if no AwardNumber)
```
Comment (progval) [inline crosswalks/spase-v2.6.1.csv:69]:
```suggestion
funding,ResourceHeader/Funding/AwardNumber or ResourceHeader/Funding/Project
```

Comment (dgarijo) [reply crosswalks/spase-v2.6.1.csv:69]: @BaptisteCecconi can you please tell us if you are ok with this suggestion, so @progval can merge the PR?

**Comment by dgarijo** [comment]
@BaptisteCecconi ping so we can finalize this PR?

**Comment by BaptisteCecconi** [comment]
This is good to go for me.

**Pull request merged**